TTC.Deployment.AmazonWebServices/Deployer.cs

```diff
@@ -149,22 +169,36 @@ private Bundle PushBundleForSubdirectory(ApplicationSetRevision applicationSetRe
 
         private void EnsureCodeDeployRoleExists()
         {
+            if (string.IsNullOrWhiteSpace(_awsConfiguration.AssumeRoleTrustDocument))
+            {
+                return;
+            }
+
+            if (string.IsNullOrWhiteSpace(_awsConfiguration.RoleName))
+            {
+                return;
+            }
+
             try
             {
-                _iamClient.CreateRole(new CreateRoleRequest
+                var response = _iamClient.CreateRole(new CreateRoleRequest
                 {
                     RoleName = _awsConfiguration.RoleName,
                     AssumeRolePolicyDocument = File.ReadAllText(_awsConfiguration.AssumeRoleTrustDocument)
                 });
+
+                _awsConfiguration.Role = response.Role;
             }
             catch (EntityAlreadyExistsException)
             {
+                _awsConfiguration.Role = 
+                    _iamClient.GetRole(new GetRoleRequest { RoleName = _awsConfiguration.RoleName }).Role;
                 return;
             }
 
             _iamClient.PutRolePolicy(new PutRolePolicyRequest
             {
-                RoleName = _awsConfiguration.RoleName,
+                RoleName = _awsConfiguration.Role.RoleName,
                 PolicyName = "s3-releases",

```

Comment:
Is this a magic string ie you have to have a policy setup called "s3-releases"? or is it just an aws name that is well known